18-DeviceService/src/PrismSample/ViewModels/MainPageViewModel.cs

```diff
@@ -1,11 +1,27 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
-using Prism.Mvvm;
+﻿using Prism.Mvvm;
+using Prism.Services;
+using Xamarin.Forms;
 
 namespace PrismSample.ViewModels
 {
     public class MainPageViewModel : BindableBase
     {
+        private readonly IDeviceService _deviceService;
+
+        public string RuntimePlatform { get; set; }
+        public string Idiom { get; set; }
+        public string XamarinRuntimePlatform { get; set; }
+        public string XamarinIdiom { get; set; }
+
+        public MainPageViewModel(IDeviceService deviceService)
+        {
+            _deviceService = deviceService;
+
+            RuntimePlatform = _deviceService.RuntimePlatform.ToString();
+            Idiom = _deviceService.Idiom.ToString();
+            
+            XamarinIdiom = Device.Idiom.ToString();
```

Comment:
I would like to avoid the possibility of causing confusion by adding in the Xamarin way of doing things. Let's remove that part and focus only on Prism.